### PR TITLE
Add confidential computing in aks module

### PR DESCRIPTION
**Pull request by jiaweitao001**
## Issue number

#[421](https://github.com/Azure/terraform-azurerm-aks/issues/421)

## Checklist before requesting a review
- [ ] The pr title can be used to describe what this pr did in `CHANGELOG.md` file
- [ ] I have executed pre-commit on my machine
- [ ] I have passed pr-check on my machine

Thanks for your cooperation!

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -388,6 +388,14 @@ variable "cluster_name" {
   description = "(Optional) The name for the AKS resources created in the specified Azure Resource Group. This variable overwrites the 'prefix' var (The 'prefix' var will still be applied to the dns_prefix if it is set)"
 }
 
+variable "confidential_computing" {
+  type = list(object({
```
Comment (lonegunmanb) [inline variables.tf:392]: According to the [schema](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/kubernetes_addons.go#L89-L101), could we set type to `boolean` for this variable?

**Review by lonegunmanb — CHANGES_REQUESTED** [review]
Thanks @jiaweitao001, one comment.

File: variables.tf (hunk 1 above)
Comment (jiaweitao001) [reply variables.tf:392]: Yes, we can simplify this type to a `boolean`, I was wondering there might be a possibility that more attributes will be added to this `confidential_computing` in the future, use a `list` would be more scalable?

Comment (lonegunmanb) [reply variables.tf:392]: Since the `MaxItems` of this list is `1` I think in that case, an `object` would be enough for us. WDYT?

Comment (jiaweitao001) [reply variables.tf:392]: Make sense. We can just use `object` instead of `list`.

**Review by lonegunmanb — APPROVED** [review]
Thanks @jiaweitao001, LGTM!

**Pull request merged**